delocate/tests/test_delocating.py

```diff
@@ -507,3 +507,150 @@ def test_bads_report():
                   set([(LIB64A, LIBBOTH, ARCH_32),
                        (LIB64, LIB64, ARCH_32, ARCH_64),
                        (LIB32, ARCH_32)]))
+
+
+def test_ld_library_path_lookups():
+    # Test that LD_LIBRARY_PATH can be used to find libs during
+    # delocation
+    old_LD_LIBRARY_PATH = os.environ.get('LD_LIBRARY_PATH', None)
+    try:
+        # Clear out LD_LIBRARY_PATH so we can run in isolation
+        if old_LD_LIBRARY_PATH is not None:
+            del os.environ['LD_LIBRARY_PATH']
+        with InTemporaryDirectory() as tmpdir:
```

Comment:
Any way of sharing code between the test functions?  There seems to be a lot of repeated code here.

Reply:
Fixed with your nice context manager suggestion. =)